native/ex_tokenizers/src/models.rs

```diff
@@ -0,0 +1,349 @@
+use std::collections::HashMap;
+use std::path::Path;
+
+use rustler::NifTaggedEnum;
+use tokenizers::models::bpe::BpeBuilder;
+use tokenizers::models::wordlevel::WordLevelBuilder;
+use tokenizers::models::wordpiece::WordPieceBuilder;
+use tokenizers::{Model, ModelWrapper};
+
+use crate::error::ExTokenizersError;
+use crate::util::DetailValue;
+
+pub struct ExTokenizersModelRef(pub ModelWrapper);
+
+#[derive(rustler::NifStruct)]
+#[module = "Tokenizers.Model"]
+pub struct ExTokenizersModel {
+    pub resource: rustler::resource::ResourceArc<ExTokenizersModelRef>,
+}
+
+impl ExTokenizersModelRef {
+    pub fn new<T>(data: T) -> Self
+    where
+        T: Into<ModelWrapper>,
+    {
+        Self(data.into())
+    }
+}
+
+impl ExTokenizersModel {
+    pub fn new<T>(data: T) -> Self
+    where
+        T: Into<ModelWrapper>,
+    {
+        Self {
+            resource: rustler::resource::ResourceArc::new(ExTokenizersModelRef::new(data)),
+        }
+    }
+}
+
+#[rustler::nif]
+pub fn models_save(
+    model: ExTokenizersModel,
+    folder: String,
+    prefix: String,
+) -> Result<Vec<String>, ExTokenizersError> {
+    Ok(model
+        .resource
+        .0
+        .save(Path::new(&folder), Some(&prefix))?
+        .iter()
+        .map(|path| {
+            path.to_str()
+                // Unwraping here, because we are sure that pathes are valid
+                .unwrap()
+                .to_owned()
+        })
+        .collect())
+}
+
+///////////////////////////////////////////////////////////////////////////////
+/// Inspection
+///////////////////////////////////////////////////////////////////////////////
+
+#[rustler::nif]
+pub fn models_info(
+    model: ExTokenizersModel,
+) -> Result<HashMap<String, DetailValue>, ExTokenizersError> {
+    Ok(match &model.resource.0 {
+        ModelWrapper::BPE(model) => HashMap::from([
+            (
+                String::from("model_type"),
```

Comment:
In the `decoders_info`, you used the `"model_type".into()` syntax to create a String. What about change it here - in this entire file - to follow that pattern?

Reply:
Good catch